ouroboros-consensus/src/Ouroboros/Consensus/Storage/Common.hs

```diff
@@ -32,84 +31,53 @@ module Ouroboros.Consensus.Storage.Common (
 
 import           Ouroboros.Network.Block (Point (..), SlotNo, genesisPoint)
 import           Ouroboros.Network.DeltaQ (SizeInBytes)
 import           Ouroboros.Network.Point (WithOrigin (..))
 
 import           Ouroboros.Consensus.Block (IsEBB)
-import           Ouroboros.Consensus.Util.Condense
+import           Ouroboros.Consensus.Util.CBOR (decodeWithOrigin,
+                     encodeWithOrigin)
 
 {-------------------------------------------------------------------------------
   Indexing
 -------------------------------------------------------------------------------}
 
--- | Tip of the chain
-data Tip r = Tip !r | TipGen
-  deriving (Show, Eq, Functor, Foldable, Traversable, Generic, NoUnexpectedThunks)
+tipIsGenesis :: WithOrigin r -> Bool
+tipIsGenesis Origin = True
+tipIsGenesis (At _) = False
 
--- | 'TipGen' is always smaller than 'Tip'
-instance Ord r => Ord (Tip r) where
-  compare x y = case (x, y) of
-    (TipGen, TipGen) -> EQ
-    (TipGen, Tip _)  -> LT
-    (Tip _,  TipGen) -> GT
-    (Tip a,  Tip b)  -> compare a b
-
-instance Condense r => Condense (Tip r) where
-  condense TipGen  = "genesis"
-  condense (Tip r) = condense r
-
-tipIsGenesis :: Tip r -> Bool
-tipIsGenesis TipGen  = True
-tipIsGenesis (Tip _) = False
-
-tipToPoint :: Tip (Point blk) -> Point blk
-tipToPoint TipGen  = genesisPoint
-tipToPoint (Tip p) = p
+tipToPoint :: WithOrigin (Point blk) -> Point blk
+tipToPoint Origin = genesisPoint
+tipToPoint (At p) = p
 
 -- | Tip from a point
 --
--- NOTE: We really shouldn't instantate 'Tip' with 'Point'; see
+-- NOTE: We really shouldn't instantate 'WithOrigin' with 'Point'; see
 -- <https://github.com/input-output-hk/ouroboros-network/issues/1155>
-tipFromPoint :: Point blk -> Tip (Point blk)
-tipFromPoint (Point Origin) = TipGen
-tipFromPoint p              = Tip p
+tipFromPoint :: Point blk -> WithOrigin (Point blk)
+tipFromPoint (Point Origin) = Origin
+tipFromPoint p              = At p
 
 {-------------------------------------------------------------------------------
   Serialization
 -------------------------------------------------------------------------------}
 
-instance Serialise r => Serialise (Tip r) where
-  encode = encodeTip encode
-  decode = decodeTip decode
-
-encodeTip :: (r     -> Encoding)
-          -> (Tip r -> Encoding)
-encodeTip encodeR tip =
-    case tip of
-      TipGen -> Enc.encodeListLen 0
-      Tip r  -> Enc.encodeListLen 1 <> encodeR r
-
-decodeTip :: (forall s. Decoder s r)
-          -> (forall s. Decoder s (Tip r))
-decodeTip decodeR = do
-    tag <- Dec.decodeListLen
-    case tag of
-      0 -> return TipGen
-      1 -> Tip <$> decodeR
-      _ -> fail "decodeTip: invalid tag"
+encodeTip :: (r            -> Encoding)
+          -> (WithOrigin r -> Encoding)
+encodeTip = encodeWithOrigin
+
+decodeTip :: (forall s. Decoder s             r)
+          -> (forall s. Decoder s (WithOrigin r))
+decodeTip = decodeWithOrigin
```

Comment:
Those should be removed

Reply:
Will do so in subsequent PR, which will close #1155.